### PR TITLE
Allow graph breaks in inductor opinfo tests

### PR DESCRIPTION
**Pull request by eellison**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #105480

Previously, we would have test failures for operators which graph broke bc dynamic shape or data dependent ops. Those would appear as a failure because we were running with `nopython=True`. Those test "failures" (which is expected behavior) obfuscated the actual correctness errors, and made this test lower signal.

If we wanted to do something like full-op export, that should be different than inductor opinfos.

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @ngimel @yf225 @chenyang78 @kadeng @muchulee8 @aakhundov

### CODE REVIEW

File: test/inductor/test_torchinductor_opinfo.py [hunk 1]
```diff
@@ -504,6 +500,10 @@ def wrapper_set_seed(op, *args, **kwargs):
     "triu",
 }
 
+allow_graph_breaks = {
+    "item",
+}
+
```
Comment (voznesenskym) [inline test/inductor/test_torchinductor_opinfo.py:506]: the whole point of these tests is capturing an op without breaks.

If this is tracing `item` but... item graph breaks, isn't that an empty graph? What is the point?

Comment (eellison) [reply test/inductor/test_torchinductor_opinfo.py:506]: We're testing correctness. I don't think it's helpful to have correctness failures mixed in with expected graph breaks, it reduces the signal of the test. 

I'm happy to do a different rearrangement of the tests wrt/graph breaks if there's something else you prefer.

Comment (eellison) [reply test/inductor/test_torchinductor_opinfo.py:488]: Did new implementation. Please take a look.

Comment (desertfire) [reply test/inductor/test_torchinductor_opinfo.py:488]: Remove this as it's not used anywhere now?

**Review by desertfire — APPROVED** [review]
Fix CI errors.

**Pull request closed without merging**